addons/hr/wizard/mail_activity_schedule.py

```diff
@@ -28,6 +28,22 @@ def _compute_plan_department_filterable(self):
         for wizard in self:
             wizard.plan_department_filterable = wizard.res_model == 'hr.employee'
 
+    @api.depends('plan_date', 'plan_id')
+    def _compute_plan_summary(self):
+        if not self.env.context.get('sort_by_responsible', False):
+            return super()._compute_plan_summary()
+        self.plan_summary = False
+        responsible_value_to_label = dict(self.env['mail.activity.plan.template']._fields['responsible_type'].selection)
+        for scheduler in self:
+            templates_by_responsible_type = defaultdict(list)
+            for template in scheduler.plan_id.template_ids:
+                responsible = responsible_value_to_label[template.responsible_type]
+                templates_by_responsible_type[responsible].append(template)
+            scheduler.plan_summary = Markup(''.join('<ul>%(responsible)s %(block)s</ul>' % {
+                'responsible': key,
+                'block': scheduler._get_summary_lines(templates)
+            } for key, templates in templates_by_responsible_type.items()))
```

Comment:
Also I might be missing something but doesn't it seems like this could just be part of the original compute method and just use a condition on "there is more than one responsible"? And just always do this?

Reply:
but the responsible type is added in hr module not in mail module.

Reply:
Yes You're right the selection_add tricked me :(

Reply:
> Also I might be missing something but doesn't it seems like this could just be part of the original compute method and just use a condition on "there is more than one responsible"? And just always do this?

for this I don't know, it's linked to mail module, and we just want this feature for hr application, i don't know if they want it too